packages/near-fast-auth-signer/src/components/SocialLogin/SocialButton.tsx

```diff
@@ -0,0 +1,147 @@
+import { createKey, isPassKeyAvailable } from '@near-js/biometric-ed25519';
+import { FacebookAuthProvider, GoogleAuthProvider, signInWithPopup } from 'firebase/auth';
+import React from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { setAccountIdToController } from '../../lib/controller';
+import FirestoreController from '../../lib/firestoreController';
+import { decodeIfTruthy } from '../../utils';
+import { networkId } from '../../utils/config';
+import { firebaseAuth } from '../../utils/firebase';
+import { generateRandomString } from '../../utils/string';
+import { onCreateAccount, onSignIn } from '../AuthCallback/AuthCallback';
+
+const Button = styled.button`
+  font-size: 14px;
+  font-weight: 600;
+  color: var(--sand12);
+  background: var(--sand3);
+  padding: 8px;
+  border: 1px solid var(--sand6);
+  display: flex;
+  align-items: center;
+  width: calc(50% - 4px) !important;
+  justify-content: center;
+  border-radius: 100px;
+  box-sizing: border-box;
+
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+
+  ${(props) => (props.disabled
+    && `
+    cursor: not-allowed;
+    opacity: 0.5;
+    color: #72727A;
+    `
+  )};
+
+`;
+
+type SocialButtonProps = {
+  provider: GoogleAuthProvider | FacebookAuthProvider;
+  logoComponent: React.ReactNode;
+  label: string;
+  isRecovery: boolean;
+  disabled?: boolean;
+};
+
+const shouldCreateAccount = (createdAt: string, lastLoginAt: string): boolean => {
+  const createdTime = new Date(createdAt).getTime();
+  const lastLoginTime = new Date(lastLoginAt).getTime();
+
+  const diffInMilliseconds = lastLoginTime - createdTime;
+  if (diffInMilliseconds === 0) return true;
+
+  const diffInMinutes = diffInMilliseconds / (1000 * 60);
+
+  return diffInMinutes < 1;
+};
+
+function SocialButton({
+  provider,
+  logoComponent,
+  label,
+  isRecovery,
+  disabled,
+}: SocialButtonProps) {
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const onClick = () => {
+    signInWithPopup(firebaseAuth, provider)
+      .then(async (result) => {
+        const { user } = result;
+        // TODO: construct accountId, confirm with UX
+        const accountId = `${generateRandomString(8)}.${networkId === 'testnet' ? 'testnet' : 'near'}`;
+        const { email } = user;
+
+        const success_url = decodeIfTruthy(searchParams.get('success_url'));
+        const public_key_lak = decodeIfTruthy(searchParams.get('public_key_lak'));
+        const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
+        const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
+        const accessToken = await user.getIdToken();
+
+        setAccountIdToController({ accountId, networkId });
+
+        let publicKeyFak: string;
+        if (await isPassKeyAvailable()) {
```

Comment:
I see the code here share a lot of similarities with AuthCallback page. Can we create functions and re-use them to avoid maintaining the same code on two parts? 

https://github.com/near/fast-auth-signer/blob/8d2726f5fe007d7ae011dec245692ac7499aff0c/packages/near-fast-auth-signer/src/components/AuthCallback/AuthCallback.tsx#L238